### ISSUE TITLE
DefaultLambdaJsonSerializer throws when deserializing object with initialized HashSet

### ISSUE THREAD

**Issue by emahnetmore** [aws/aws-lambda-dotnet#1037]
## Description
Deserializing an object of a class that contains an initialized `HashSet` with `DefaultLambdaJsonSerializer` throws a `Amazon.Lambda.Serialization.SystemTextJson.JsonSerializerException`.
I'm expecting no exception to be thrown. Deserializing the exact same thing using regular `System.Text.Json.JsonSerializer` works fine.

## Reproduction Steps
Run the following program:
```cs
public class Example
{
    public HashSet<int> Integers { get; set; } = new();
}

public class Program
{
    public static async Task Main()
    {
        var stream = new MemoryStream();
        var example = new Example();
        await JsonSerializer.SerializeAsync(stream, example);

        // This throws. Same error whichever constructor is used.
        stream.Position = 0;
        var defaultLambdaJsonSerializer = new DefaultLambdaJsonSerializer();
        Example? awsExample = defaultLambdaJsonSerializer.Deserialize<Example>(stream);

        // This works fine.
        stream.Position = 0;
        Example? regularStjExample = await JsonSerializer.DeserializeAsync<Example>(stream);
    }
}
```
Stacktrace:
```
Amazon.Lambda.Serialization.SystemTextJson.JsonSerializerException
  HResult=0x80131500
  Message=Error converting the Lambda event JSON payload to type TestingDefaultSystemTextJsonSerializer.Example: Unable to cast object of type 'System.Collections.Generic.HashSet`1[System.Int32]' to type 'System.Collections.IList'.
  Source=Amazon.Lambda.Serialization.SystemTextJson
  StackTrace:
   at Amazon.Lambda.Serialization.SystemTextJson.AbstractLambdaJsonSerializer.Deserialize[T](Stream requestStream)
   at TestingDefaultSystemTextJsonSerializer.Program.<Main>d__0.MoveNext() in C:\Users\work\source\repos\TestingDefaultSystemTextJsonSerializer\TestingDefaultSystemTextJsonSerializer\Program.cs:line 21

  This exception was originally thrown at this call stack:
    [External Code]

Inner Exception 1:
InvalidCastException: Unable to cast object of type 'System.Collections.Generic.HashSet`1[System.Int32]' to type 'System.Collections.IList'.

```
The same error occurs if using any collection that cannot be cast into an `IList`, such as `System.Net.CookieCollection`.
Removing the initialization of the `HashSet`/`Collection` fixes the issue, i.e:
```cs
public class Example
{
    public HashSet<int> Integers { get; set; }
}
```

### Nb!
Tested on dotnetcore3.1, net5.0 and net6.0.
The error **ONLY** occurs on dotnetcore3.1.
This does not seem to be an issue on net5.0 or net6.0.

### Environment
* Build Version: 2.2.0
* OS Info: Windows 11 Pro
* Build Environment: Visual Studio 2022
* Targeted .NET Platform:  dotnetcore3.1

## Resolution
- [ ] :wave: I can/would-like-to implement a fix for this problem myself

This is a :bug: bug-report

**Comment by ashishdhingra** [comment]
Reproducible using target framework `netcoreapp3.1` using the below .csproj file (kindly note the use of `<LangVersion>9.0</LangVersion>` since the target types object creation is only available in C# 9.0 or later):
```XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <LangVersion>9.0</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <None Remove="Amazon.Lambda.Serialization.SystemTextJson" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Amazon.Lambda.Serialization.SystemTextJson" Version="2.2.0" />
  </ItemGroup>
</Project>
```

**Comment by ashishdhingra** [comment]
Even though the issue is reproducible in .NET 3.1 Core, `.NET Core 3.1` had reached end of life per https://dotnet.microsoft.com/en-us/download/dotnet/3.1. Based on discussion with team, closing this issue since it is not reproducible in `.NET 6` (or later).
